sqlair.go

```diff
@@ -92,15 +97,98 @@ type Iterator struct {
 	started bool
 }
 
+// A Least Recently Used Cache using a doubly linked list.
+type lruCache struct {
+	ll   *list.List
+	c    map[index]*list.Element
```

Comment:
I've changed `c` to `cache` and put some explanation comments above `ll` saying that it is a doubly linked list and that the last element is the LRU. It may still be a bit confusing but idk if the other two names can really be improved much to add clarity (i.e. `dlList` is still pretty opaque and `doublyLinkedList` is a bit long imo.

Reply:
I think it could be better to name the field after what it does rather than how it's made, perhaps `recentlyUsed` could work?